pkg/sqlite/query.go

```diff
@@ -752,3 +780,157 @@ func (s *SQLQuerier) GetCurrentCSVNameForChannel(ctx context.Context, pkgName, c
+
+	rows, err := s.db.QueryContext(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	bundles = []*api.Bundle{}
+	bundlesMap := map[string]*api.Bundle{}
+	for rows.Next() {
+		var entryID sql.NullInt64
+		var bundle sql.NullString
+		var bundlePath sql.NullString
+		var minDepth sql.NullInt64
+		var bundleName sql.NullString
+		var pkgName sql.NullString
+		var channelName sql.NullString
+		var replaces sql.NullString
+		var version sql.NullString
+		var skipRange sql.NullString
+		var depType sql.NullString
+		var depPkgName sql.NullString
+		var depGroup sql.NullString
+		var depVersion sql.NullString
+		var depKind sql.NullString
+		if err := rows.Scan(&entryID, &bundle, &bundlePath, &minDepth, &bundleName, &pkgName, &channelName, &replaces, &version, &skipRange, &depType, &depPkgName, &depGroup, &depVersion, &depKind); err != nil {
+			return nil, err
+		}
+
+		bundleKey := fmt.Sprintf("%s/%s/%s", bundleName.String, version.String, bundlePath.String)
+		bundleItem, ok := bundlesMap[bundleKey]
+		if ok {
+			// Create new dependency object
+			dep := &api.Dependency{}
+			dep.Type = depType.String
+			dep.Name = depPkgName.String
+			dep.Group = depGroup.String
+			dep.Version = depVersion.String
+			dep.Kind = depKind.String
+
+			// Add new dependency to the existing list
+			existingDeps := bundleItem.Dependencies
+			existingDeps = append(existingDeps, dep)
+			bundleItem.Dependencies = existingDeps
+		} else {
+			// Create new bundle
+			out := &api.Bundle{}
+			if bundle.Valid && bundle.String != "" {
+				out, err = registry.BundleStringToAPIBundle(bundle.String)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			out.CsvName = bundleName.String
+			out.PackageName = pkgName.String
+			out.ChannelName = channelName.String
+			out.BundlePath = bundlePath.String
+			out.Version = version.String
+			out.SkipRange = skipRange.String
+
+			provided, required, err := s.GetApisForEntry(ctx, entryID.Int64)
+			if err != nil {
+				return nil, err
+			}
+			out.ProvidedApis = provided
+			out.RequiredApis = required
+
+			// Create new dependency and dependency list
+			dep := &api.Dependency{}
+			dependencies := []*api.Dependency{}
+			dep.Type = depType.String
+			dep.Name = depPkgName.String
+			dep.Group = depGroup.String
+			dep.Version = depVersion.String
+			dep.Kind = depKind.String
+			dependencies = append(dependencies, dep)
+			out.Dependencies = dependencies
+
+			bundlesMap[bundleKey] = out
+		}
+	}
+
+	for _, v := range bundlesMap {
+		if len(v.Dependencies) > 1 {
+			newDeps := unique(v.Dependencies)
+			v.Dependencies = newDeps
+		}
+		bundles = append(bundles, v)
+	}
+
+	return
+}
+
+func unique(deps []*api.Dependency) []*api.Dependency {
+	keys := make(map[string]bool)
+	list := []*api.Dependency{}
+	for _, entry := range deps {
+		depKey := fmt.Sprintf("%s/%s/%s/%s/%s", entry.Type, entry.Name, entry.Group, entry.Version, entry.Kind)
```

Comment:
this shouldn't be needed, you can use entry as a key directly

Reply:
Using a pointer as a key is not helpful for key lookup in this case. Not sure if I missed something but entry is a pointer.